src/session/query.c

```diff
@@ -22,6 +22,15 @@
 #include "zenoh-pico/session/resource.h"
 #include "zenoh-pico/utils/logging.h"
 
+_z_reply_t *_z_reply_new(_z_reply_t _reply)
+{
+    _z_reply_t *reply = (_z_reply_t *)z_malloc(sizeof(_z_reply_t));
+    if (reply != NULL) {
+        *reply = _reply;
+    }
+    return reply;
+}
+
```

Comment:
This might be error prone if reply is cleared it will affect the new reply.
I think here we should use the smart pointer helpers that @Mallets implemented some time ago.

Reply:
Could you please point me some sample of these smart pointers usage?